e2etest/l2tests/assignment.go

```diff
@@ -200,13 +204,24 @@ var _ = ginkgo.Describe("IP Assignment", func() {
 					AllocateTo: &metallbv1beta1.ServiceAllocation{Priority: 20, Namespaces: []string{f.Namespace.Name}},
 				},
 			}
+			testNs, err := cs.CoreV1().Namespaces().Get(context.Background(), f.Namespace.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+			if testNs.Labels == nil {
+				testNs.Labels = map[string]string{}
+			}
+			updateLabels(testNs.Labels)
+			_, err = cs.CoreV1().Namespaces().Update(context.Background(), testNs, metav1.UpdateOptions{})
+			framework.ExpectNoError(err)
+			nsLabelSelector := make(map[string]string)
```

Comment:
same, just add the map to the selector below. I think the result will be clearer even if it adds repetition on the labels to be added